nvflare/fuel/hci/client/api.py

```diff
@@ -164,9 +170,56 @@ def _load_client_cmds(self, cmd_modules):
     def register_command(self, cmd_entry):
         self.all_cmds.append(cmd_entry.name)
 
+    def start_session_monitor(self, session_ended_callback, interval=5):
+        if self.sess_monitor_thread and self.sess_monitor_thread.is_alive():
+            self.close_session_monitor()
+        self.sess_monitor_thread = threading.Thread(
+            target=self._check_session, args=(session_ended_callback, interval), daemon=True
+        )
+        self.sess_monitor_active = True
+        self.sess_monitor_thread.start()
+
+    def close_session_monitor(self):
+        self.sess_monitor_active = False
+        if self.sess_monitor_thread:
```

Comment:
Change to:
`if self.sess_monitor_thread and self.sess_monitor_thread.is_alive():`